### ISSUE TITLE
Connection is unbreakable

### ISSUE THREAD

**Issue by Enleur** [php-amqplib/php-amqplib#368]
On 2.6.0 version there is a bug. Timeouts for stream/socket connection aren't working.
For instance, use https://github.com/php-amqplib/php-amqplib/blob/master/demo/amqp_consumer.php to connect to a VM with a rabbitmq server and after consuming couple of messages turn off network between the client and the VM. The stream will be hanging even after restoring the network connection. 

Tested with turned on heartbeat option and with the heartbeat(timeouts 20s, heartbeat 10s).

In 2.5.0 version if the heartbeat turned on - after timeout connection will be dropped by timeout, if turned off - it waits until connection goes back and consumes ok.

Also I had on production server high CPU usage by idle consumers on 2.6.0 version, but cannot reproduce on local machine.

**Comment by postalservice14** [comment]
I've tried, and I can't seem to reproduce this.  Here's what I did.

1. Start a VM with Rabbit
2. Start the `amqp_consumer.php` demo script (except using the connection info below)
3. Suspend the VM

What I expect and DOES happen:
I eventually get an AMQPRuntimeException with message 'Broken pipe or closed connection'

Is that not what you are seeing?

__My Connection__
```php
$conn = new AMQPStreamConnection(
    HOST,
    PORT,
    USER,
    PASS,
    VHOST,
    false,
    'AMQPLAIN',
    null,
    'en_US',
    3,
    20,
    null,
    false,
    10
);
```
